lib/Dialect/FIRRTL/Transforms/DropName.cpp

```diff
@@ -0,0 +1,80 @@
+//===- DropName.cpp - Drop Names  -----------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the DropName pass.
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetails.h"
+#include "circt/Dialect/FIRRTL/FIRRTLOps.h"
+#include "circt/Dialect/FIRRTL/Passes.h"
+#include "llvm/ADT/STLExtras.h"
+
+using namespace mlir;
+using namespace circt;
+using namespace firrtl;
+
+static bool isDeadOp(Operation *op,
+                     llvm::SmallDenseSet<Operation *> &deadOperations) {
+  for (auto *user : op->getUsers()) {
+    if (deadOperations.count(user))
+      continue;
+
+    auto connect = dyn_cast<FConnectLike>(user);
+    // If the user is neither node nor connect, we consider the op to be alive.
+    if (!connect)
+      return false;
+    if (connect.src().getDefiningOp() != op)
```

Comment:
This only works for type-lowered code, yes?

Reply:
Changed to reject non-lowered types explicitly.